src/input.rs

```diff
@@ -12,33 +14,51 @@ fn matches_key(key: Key, acceptable_keys: &[String]) -> bool {
     }
 }
 
-pub fn input_loop(mut controller: Controller) -> Result<(), Box<dyn Error>> {
-    controller.on_startup()?;
+pub fn input_loop(controller: Controller) -> Result<(), Box<dyn Error>> {
+    let keybinding = controller.config.keybinding.clone();
+
+    let am_controller = Arc::new(Mutex::new(controller));
+    am_controller.lock().unwrap().on_startup()?;
+
     let stdin = stdin();
-    let keybinding = &controller.config.keybinding.clone();
 
     for c in stdin.keys() {
         match c {
             Ok(key) => {
                 if matches_key(key, &keybinding.quit) {
-                    controller.on_keypress_quit()?;
+                    am_controller.lock().unwrap().on_keypress_quit()?;
                     break;
                 } else if matches_key(key, &keybinding.down) {
-                    controller.on_keypress_down()?;
+                    am_controller.lock().unwrap().on_keypress_down()?;
                 } else if matches_key(key, &keybinding.up) {
-                    controller.on_keypress_up()?;
+                    am_controller.lock().unwrap().on_keypress_up()?;
                 } else if matches_key(key, &keybinding.start) {
-                    controller.on_keypress_start()?;
+                    if let Some((pid, process_index)) = am_controller.lock().unwrap().on_keypress_start()? {
+                        watch_pid(am_controller.clone(), pid, process_index);
+                    }
                 } else if matches_key(key, &keybinding.stop) {
-                    controller.on_keypress_stop()?;
+                    am_controller.lock().unwrap().on_keypress_stop()?;
                 }
             },
             Err(e) => {
-                controller.on_error(Box::new(e));
+                am_controller.lock().unwrap().on_error(Box::new(e));
             }
         }
     }
 
-    controller.on_exit()?;
+    am_controller.lock().unwrap().on_exit()?;
     Ok(())
 }
+
+fn watch_pid(controller: Arc<Mutex<Controller>>, pid: i32, process_index: usize) {
+    spawn(move || unsafe {
+        let mut file = std::fs::File::create("foo.txt").unwrap();
+        use std::io::prelude::*;
+        let l1 = format!("{}\n", pid);
+        let _ = file.write_all(l1.as_bytes());
+        // BUG: waitpid returns immediately, should options be something other than 0?
```

Comment:
This probably doesn't work since the process is a child of tmux, not proctmux

Will look into how to wait for any pid

Reply:
Tmux control mode might be an appropriate solution to listen for when panes die:

https://github.com/tmux/tmux/wiki/Control-Mode